Wikipedia/Code/WMFSearchResults.m

```diff
@@ -96,8 +96,13 @@ - (void)mergeRedirectMappingsFromModel:(WMFSearchResults *)searchResults {
 }
 
 - (void)mergeResultsFromModel:(WMFSearchResults *)searchResults {
+    NSMutableArray *displayTitlesOfExistingResults = [NSMutableArray new];
```

Comment:
This should be a `NSMutableSet` so that the lookup below is constant time instead of linearly increasing with the size of `displayTitlesOfExistingResults`.